src/app/rpmostree-clientlib.cxx

```diff
@@ -1244,6 +1244,82 @@ rpmostree_print_diff_advisories (GVariant *rpm_diff, GVariant *advisories, gbool
   return TRUE;
 }
 
+/* print "manifest-diff" */
+gboolean
+rpmostree_print_manifest_diff (GVariant *manifest_diff, guint maxkeylen, GError **error)
+{
+  int total_str = 0;
+  int total_size_str = 0;
+  int total_removed_str = 0;
+  int removed_size_str = 0;
+  int total_added_str = 0;
+  int added_size_str = 0;
+
+  g_auto (GVariantDict) manifest_diff_dict;
+  g_variant_dict_init (&manifest_diff_dict, manifest_diff);
+
+  g_autoptr (GVariant) total
+      = g_variant_dict_lookup_value (&manifest_diff_dict, "total", G_VARIANT_TYPE ("t"));
+  if (!total)
+    return FALSE;
```

Comment:
In the GError rules, if `FALSE` is returned then `error` must be set.  So...here and everywhere else we'd need to do:

```
if (!total)
  return glnx_throw (error, "failed to find total");
```

---

However...if it "can't happen" that we're missing the keys, we could also just `g_assert (total);` which may be simpler.